test/sender_and_receiver_integration_test.exs

```diff
@@ -114,6 +115,36 @@ defmodule SenderAndReceiverIntegrationTest do
     end
   end
 
+  describe "timeout behaviour" do
+    test "does not reconnect when a receive timeout occurs" do
```

Comment:
Do you think, this test description now should change to "does not add additional process/leak ports on reconnect"

Reply:
And I think if you't use `SenderAndReceiverIntegrationTest.TestSlowDispatcher` in test, it will still work same way because message never ends well and it times out due to very low timeout.

we can also remove `TestNoReplyDispatcher`, we are not using it anywhere

Reply:
Can change the description, n/p. 

>  And I think if you't use SenderAndReceiverIntegrationTest.TestSlowDispatcher in test, it will still work same way because message never ends well and it times out due to very low timeout. we can also remove TestNoReplyDispatcher, we are not using it anywhere

Once again, good catch. That's also a bug though. That is, if I'm specifying a module that doesn't exist and it doesn't result in a crash, it's a bug. Will open an issue for this. 


If you change sender.ex back to the version from `main` and remove the `Process.sleep/1` call you'll see it passes when it shouldn't

Reply:
yes, It passes on main as expected but fails if we add `Process.sleep/1` which is expected because we don't close port before attempting the connection again